src/app_service/service/community/service.nim

```diff
@@ -244,9 +244,22 @@ QtObject:
 
   proc leaveCommunity*(self: Service, communityId: string) =
     try:
-      discard status_go.leaveCommunity(communityId)
+      let response = status_go.leaveCommunity(communityId)
+
+      if response.error != nil:
+        let error = Json.decode($response.error, RpcError)
+        raise newException(RpcException, "Error leaving community: " & error.message)
+
+      if response.result == nil or response.result.kind == JNull:
+        error "error: ", methodName="leaveCommunity", errDesription = "result is nil"
+        return
+
+      let community = response.result["communities"][0].toCommunityDto()
```

Comment:
This is not necessary. You already have the ID of the community from the function args.

Reply:
@Khushboo-dev-cpp this is a nice optimization, since we actually just need to need to know if `status_go.leaveCommunity(communityId)` call passed with success or not, and the `communityId` is already known. Maybe one day we may improve `status-go` and actually update a response of this call to return just if action passed with success or not.

Reply:
I agree, I see a lot of responses and events for communities especially not optimised. it will be visible in my next PR for updating on channel addition removal from admins side. Or maybe I knowledge is incomplete. Willl do :)